python/lsst/obs/base/ingest_tests.py

```diff
@@ -109,33 +106,55 @@ def instrumentName(self):
         """
         return self.instrumentClass.getName()
 
-    def setUp(self):
+    @classmethod
+    def setUpClass(cls):
         # Use a temporary working directory
-        self.root = tempfile.mkdtemp(dir=self.ingestDir)
-        self._createRepo()
+        cls.root = tempfile.mkdtemp(dir=cls.ingestDir)
+        cls._createRepo()
 
         # Register the instrument and its static metadata
-        self._registerInstrument()
+        cls._registerInstrument()
 
-    def tearDown(self):
-        if os.path.exists(self.root):
-            shutil.rmtree(self.root, ignore_errors=True)
+    def setUp(self):
+        # Want a unique run name per test
+        self.outputRun = "raw_ingest_" + str(random.randrange(1_000_000_000))
 
-    def verifyIngest(self, files=None, cli=False):
+    @classmethod
+    def tearDownClass(cls):
+        if os.path.exists(cls.root):
+            shutil.rmtree(cls.root, ignore_errors=True)
+
+    def verifyIngest(self, files=None, cli=False, fullCheck=False):
         """
         Test that RawIngestTask ingested the expected files.
 
         Parameters
         ----------
         files : `list` [`str`], or None
             List of files to be ingested, or None to use ``self.file``
+        fullCheck : `bool`, optional
+            If `True`, read the full raw dataset and check component
+            consistency. If `False` check that a component can be read
+            but do not read the entire raw exposure.
         """
         butler = Butler(self.root, run=self.outputRun)
-        datasets = butler.registry.queryDatasets(self.outputRun, collections=...)
+        datasets = butler.registry.queryDatasets("raw", collections=self.outputRun)
         self.assertEqual(len(list(datasets)), len(self.dataIds))
+
+        # Reading all the ingested test data can be expensive. The code paths
+        # for reading the second raw are the same as reading the first so
+        # we do not gain anything by doing full checks of everything.
+        # Only read full pixel data for first dataset from file.
+        # Don't even do that if we are requested not to by the caller.
+        # This only really affects files that contain multiple datasets.
```

Comment:
I think this comment block should go into either a long description in the method docstring, or a `Notes` section therein: it's rather hidden in here.